htdocs/class/smarty/xoops_plugins/function.block.php

```diff
@@ -43,7 +43,7 @@ function smarty_function_block($params, &$smarty)
 
     static $block_objs;
     if (!isset($block_objs[$block_id])) {
-        include_once XOOPS_ROOT_PATH . '/class/xoopsblock.php';
+        include_once XOOPS_ROOT_PATH . '/kernel/block.php';
```

Comment:
The proper replacement should be:  `xoops_getHandler('block');`

That way we stop loading files from specific locations, and use the systems internal knowledge to do the job for us!